### PR TITLE
Properly pack @markprompt/react

### PR DESCRIPTION
**Pull request by remcohaszing**
The `markprompt` package was renamed to `@markprompt/react`. It is now ESM-only, and TypeScript is configured accordingly.

The package directory was renamed from `markprompt-react` to `react`. This allows us to map paths using a glob when adding more packages.

The package is now built using `tsc --build` instead of a custom build script.

### CODE REVIEW

File: packages/react/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "include": ["src"],
+  "compilerOptions": {
+    "module": "node16",
+    "declaration": true,
```
Comment (nickrttn) [inline packages/react/tsconfig.json:5]:
```suggestion
    "declaration": true,
    "declarationMap": true,
```

File: packages/react/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "sourceMap": true,
+    "rootDir": "./src",
+    "strict": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
```
Comment (nickrttn) [inline packages/react/tsconfig.json:13]: I actually use that sometimes (although no use case here yet) and it's kind of nice? 😬

File: packages/react/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "paths": {
+      "@markprompt/*": ["../*/dist"]
```
Comment (remcohaszing) [inline packages/react/tsconfig.json:15]:
```suggestion
      "@markprompt/*": ["../*/src"]
```

File: packages/react/tsconfig.json (hunk 2 above)
Comment (nickrttn) [reply packages/react/tsconfig.json:13]: Then lets just keep `strict` and remove the rest?

Comment (remcohaszing) [reply packages/react/tsconfig.json]: Fine with me :+1:

**Review by nickrttn — APPROVED** [review]
🚀

**Pull request merged**